### PR TITLE
Add reduceProps

### PR DESCRIPTION
**Pull request by istarkov**
It looks like it's very useful decorator for my work, 
It also solve this issue https://github.com/acdlite/recompose/issues/8
And this https://github.com/acdlite/recompose/issues/18

### CODE REVIEW

**Comment by acdlite** [comment]
Thanks for the suggestion. Let me think about it for a bit.

File: src/reduceProps.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import curry from 'lodash/function/curry';
+import wrapDisplayName from './wrapDisplayName';
+import createElement from './createElement';
+
+const reduceProps = (reduce, initialState, BaseComponent) => {
+  let state = initialState;
```
Comment (acdlite) [inline src/reduceProps.js]: This value is shared between all instances of ReduceProps. You need it to be local to each component. The only way to do this is with a class component.

Comment (istarkov) [inline src/reduceProps.js]: Ooops, what a stupid bug I made :-(

**Comment by istarkov** [comment]
Look's like I need to rethink this.

**Pull request closed without merging**